app/main/views/buyers.py

```diff
@@ -29,3 +32,35 @@ def find_buyer_by_brief_id():
         title=title,
         brief_id=brief_id
     )
+
+
+@main.route('/buyers/add-buyer-domains', methods=['GET', 'POST'])
+@role_required('admin')
+def add_buyer_domains():
+    email_domain_form = EmailDomainForm()
+    status_code = 200
+    if email_domain_form.validate_on_submit():
+        try:
+            new_domain = request.form.get('new_buyer_domain')
+            data_api_client.create_buyer_email_domain(new_domain, current_user.email_address)
+            flash('You’ve added {}.'.format(new_domain), 'message')
+            return redirect(url_for('.add_buyer_domains'))
+        except HTTPError as e:
+            status_code = 400
+            if "has already been approved" in e.message:
+                email_domain_form.new_buyer_domain.errors.append(
+                    'You cannot add this domain because it already exists.'
+                )
+            elif "was not a valid format" in e.message:
+                email_domain_form.new_buyer_domain.errors.append(
+                    "‘{}’ is not a valid format.".format(new_domain)
+                )
+            else:
+                raise e
+    elif email_domain_form.new_buyer_domain.errors:
```

Comment:
Why do we need to check if there are `email_domain_form.new_buyer_domain.errors` here? If `email_domain_form.validate_on_submit()` fails, don't we want to return a 400 status whether or not there's an error on that particular form field (i.e. just do an `else:` on this line)?

Reply:
It's for the `GET` route - we want to return a 200 for the empty form page on a `GET`.

Reply:
Ahaaa of course, this view has both methods doesn't it? 😸  I'd probably still go for the more generic `elif email_domain_form.errors`, but this form is unlikely to get more complicated in the future so it's probably fine as is.